### PR TITLE
Remove netstandard targets from asp.netcore instrumentation

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #.

## Changes

As discussed during SIG meeting, removing `netstandard2.0` and `netstandard2.1` targets from the library. 

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.client.cs [hunk 1]
```diff
@@ -23,7 +23,9 @@
 using Greet;
 using Grpc.Core;
 using Grpc.Net.Client;
+#if NET6_0_OR_GREATER
```
Comment (alanwest) [inline test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.client.cs:26]: Further down in this file we have a `#if !NETFRAMEWORK` directive.

https://github.com/open-telemetry/opentelemetry-dotnet/pull/3567/files#diff-9dafa45d9de8176d95436fabb3fcead50051f75d519bc1f4b99d782ad53962afL120-L126

I think it makes sense to standardize on one or the other. My vote for both of them to be `NET6_0_OR_GREATER`.

**Review by alanwest — APPROVED** [review]
LGTM

**Comment by vishweshbankwar** [comment]
@utpilla - Can we merge this one before your release?

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 2]
```diff
@@ -2,12 +2,15 @@
 
 ## Unreleased
 
+* Removed `netstandard2.0` and `netstandard2.1` targets
+  ([#3576](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3567))
```
Comment (alanwest) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:6]: Suggest making it more clear what this change means

```suggestion
* Removed `netstandard2.0` and `netstandard2.1` targets. .NET 5 reached EOL
  in May 2022 and .NET Core 3.1 reaches EOL in December 2022. The
  instrumentation for ASP.NET Core now requires .NET 6 or later.
  ([#3567](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3567))
```

**Pull request merged**